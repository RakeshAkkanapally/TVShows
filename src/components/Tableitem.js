import {
  makeStyles,
  Table,
  TableRow,
  TableHead,
  TableCell,
  TableContainer,
  TableBody,
} from "@material-ui/core";
import React from "react";
import { PropTypes } from "prop-types";

const useStyles = makeStyles((theme) => ({
  table: {
    maxHeight: theme.spacing(50),
  },
}));

const TableItem = ({ headerRow, innerHTMLRow, rows }) => {
  const classes = useStyles();
  return (
    <TableContainer>
      <Table className={classes.table} aria-label="table">
        <TableHead>
          <TableRow>
            <TableCell align="left">
              <h3>{headerRow.name}</h3>
            </TableCell>
            <TableCell align="left">
              <h3>{headerRow.value}</h3>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows &&
            rows.map((row) => (
              <TableRow key={row.name}>
                <TableCell align="left">{row.name}</TableCell>
                <TableCell align="left">{row.value}</TableCell>
              </TableRow>
            ))}
          {innerHTMLRow &&
            innerHTMLRow.map((row) => (
              <TableRow key={row.name}>
                <TableCell align="left">{row.name}</TableCell>
                <TableCell align="left">
                  <div
                    dangerouslySetInnerHTML={{
                      __html: row.value,
                    }}
                  ></div>
                </TableCell>
              </TableRow>
            ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

TableItem.propTypes = {
  headerRow: PropTypes.object,
  innerHTMLRow: PropTypes.array,
  rows: PropTypes.array,
};
export default TableItem;
