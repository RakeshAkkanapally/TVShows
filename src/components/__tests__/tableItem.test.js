import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Tableitem from "./../Tableitem";

describe("verifying TableItem component",()=>{

  it("TableItem component renders correctly", () => {
    let props={
      "headerRow": { name: "Title", value: "Title" },
      "rows": [{ name: "row", value:"value" }],
      "innerHTMLRow": [{ name: "Summary", value:"summary" }]
    };

    const tree = renderer
      .create(
        <BrowserRouter>
          <Tableitem {...props}/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("TableItem renders correctly", () => {
    let props={
      "headerRow": { name: "Title", value: "Title" },
      "rows": [{ name: "Row", value: "Title" }],
      "innerHTMLRow": [{ name: "Summary", value: "Summary" }]
    };

    const tree = renderer
      .create(
        <BrowserRouter>
          <Tableitem {...props}/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  
});