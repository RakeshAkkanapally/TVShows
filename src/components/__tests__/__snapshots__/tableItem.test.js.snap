// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verifying TableItem component TableItem component renders correctly 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    aria-label="simple table"
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
          scope="col"
        >
          <h3>
            Title
          </h3>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
          scope="col"
        >
          <h3>
            Title
          </h3>
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
        >
          row
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
        >
          value
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
        >
          Summary
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "summary",
              }
            }
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`verifying TableItem component TableItem renders correctly 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    aria-label="simple table"
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
          scope="col"
        >
          <h3>
            Title
          </h3>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
          scope="col"
        >
          <h3>
            Title
          </h3>
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
        >
          Row
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
        >
          Title
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
        >
          Summary
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "Summary",
              }
            }
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
